src/voxel.h

```diff
@@ -413,10 +413,21 @@ class VoxelManipulator /*: public NodeContainer*/
 	}
 	// Stuff explodes if non-emerged area is touched with this.
 	// Emerge first, and check VOXELFLAG_NO_DATA if appropriate.
-	MapNode & getNodeRefUnsafe(v3s16 p)
+	MapNode & getNodeRefUnsafe(const v3s16 &p)
 	{
 		return m_data[m_area.index(p)];
 	}
+
+	MapNode & getNodeRefUnsafeCheckFlags(const v3s16 &p)
+	{
+		s32 index = m_area.index(p);
+
+		if (m_flags[index] & VOXELFLAG_NO_DATA)
+			return ContextIgnoreNode;
```

Comment:
Just commenting on what I said on IRC.

This is not safe (and not desirable, IMO). The issue is that the static member `ContextIgnoreNode` is returned as a non-const object reference. Therefore anything or anyone using this function can change `ContextIgnoreNode` and cause mayhem.